esp/services/esdl_svc_engine/esdl_store.cpp

```diff
@@ -0,0 +1,865 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2013 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "jexcept.hpp"
+#include "jsmartsock.hpp"
+#include "dasds.hpp"
+#include "daclient.hpp"
+#include "dalienv.hpp"
+#include "dadfs.hpp"
+#include "dasess.hpp"
+#include "dautils.hpp"
+#include "wsexcept.hpp"
+#include "httpclient.hpp"
+#include "espcontext.hpp"
+#include "esdl_store.hpp"
+#include "esdl_binding.hpp"
+#include <memory>
+
+static const char* ESDL_DEFS_ROOT_PATH="/ESDL/Definitions/";
+static const char* ESDL_DEF_PATH="/ESDL/Definitions/Definition";
```

Comment:
It's my understanding that the major performance gain would come from changing the Definitions to be uniquely identifiable based on the element name... 
ie
/ESDL/Definitions/Definition @id=mathservice.1 would turn to 
/ESDL/Definitions/mathservice.1

two points... 
Although that is the one performance improvement suggested by @jakesmith, I'm not sure if the performance improvement potential is high enough to pursue or not, but I do believe that's what Jake pointed out.
Also, I'm not sure if now we need to set a nomenclature rule for the definition names. in other words, I don't know if allowing the Definitions to be named freely by the developer is ok or not, or if we need to come up w/ a standard naming convention like the workunits... /Workunits/W<timestamp>

Reply:
@rpastrana the scope of this issue is to create the interfaces and encapsulate the Dali operations. There's another issue for the performance improvement(HPCC-18492). The 2 concerns you pointed out are very good points, I'll copy over the comments to the other issue, and we'll discuss further.